File: CSharp/Services/ComponentAService.cs
namespace CSharpDi.Services
{
    public class ComponentAService
    {
        private readonly Logger logger;
        private readonly SingletonState singletonState;
        private readonly TransientState transientState;

        public ComponentAService(Logger logger, SingletonState singletonState, TransientState transientState)
        {
            this.logger = logger;
            this.singletonState = singletonState;
            this.transientState = transientState;
        }

        public void DoIt()
        {
            logger.LogInformation($"ComponentAService (SingletonState) {singletonState.GetValue()}");
            logger.LogInformation($"ComponentAService (TransientState) {transientState.GetValue()}");
        }
    }
}

File: Python/services/ComponentAService.py
from kink import inject
from services.SingletonState import SingletonState
from services.TransientState import TransientState
from services.logger import Logger

@inject
class ComponentAService():
    def __init__(self, logger: Logger, transient_state: TransientState, singleton_state: SingletonState):
        self._logger = logger
        self._transient_state = transient_state
        self._singleton_state = singleton_state

    def do_it(self):
        self._logger.log_information(f"ComponentAService (TransientState): {self._transient_state.get_value()}")
        self._logger.log_information(f"ComponentAService (SingletonState): {self._singleton_state.get_value()}")
        